### PR TITLE
gnutls: remove Guile bindings, guile-gnutls: init at 3.7.11

### PR DESCRIPTION
**Pull request by foo-dogsquared**
###### Description of changes

[GnuTLS has dropped its Guile bindings since 3.8.0](https://gitlab.com/gnutls/gnutls/-/blob/3.8.0/NEWS?ref_type=tags#L45-47) and now lives in [a separate repository](https://gitlab.com/gnutls/guile/).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/guile-modules/guile-gnutls/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    "--with-guile-site-ccache-dir=${builtins.placeholder "out"}/share/guile/site"
+    "--with-guile-extension-dir=${builtins.placeholder "out"}/share/guile/extensions"
+  ];
+
+  meta = with lib; {
+    homepage = "https://gitlab.com/gnutls/guile/";
+    description = "Guile bindings for GnuTLS library";
+    license = licenses.lgpl21Plus;
+    maintainers = with maintainers; [ foo-dogsquared ];
+    platforms = platforms.gnu ++ platforms.linux;
```
Comment (SuperSandro2000) [inline pkgs/development/guile-modules/guile-gnutls/default.nix:40]: Do you mean everything that is gnu *and* linux? Right now this means everything under gnu and linux, so even linux that are none gnu.

File: pkgs/development/guile-modules/guile-gnutls/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+, pkg-config
+, texinfo
+}:
+
+stdenv.mkDerivation rec {
+  pname = "guile-gnutls";
+  version = "3.7.11";
+
+  src = fetchurl {
+    url = "mirror://gnu/gnutls/${pname}-${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/development/guile-modules/guile-gnutls/default.nix:16]:
```suggestion
    url = "mirror://gnu/gnutls/guile-gnutls-${version}.tar.gz";
```

File: pkgs/development/guile-modules/guile-gnutls/default.nix (hunk 1 above)
Comment (foo-dogsquared) [reply pkgs/development/guile-modules/guile-gnutls/default.nix:40]: I meant everything that is Linux. I just copied some other Guile module package and forgot to change it.

File: pkgs/development/guile-modules/guile-gnutls/default.nix (hunk 2 above)
Comment (foo-dogsquared) [reply pkgs/development/guile-modules/guile-gnutls/default.nix]: Done.

**Pull request merged**